website/pages/docs/k8s/connect/ingress-gateways.mdx

```diff
@@ -5,4 +5,203 @@ sidebar_title: Ingress Gateways
+
+The yaml snippet is the launching point for a valid configuration that must be supplied when installing using the [official consul-helm chart](https://hub.helm.sh/charts/hashicorp/consul).
+Information on additional options can be found in the [Helm reference](/docs/k8s/helm). Configuration options for ingress gateways reside under the [ingressGateways](/docs/k8s/helm#v-ingressgateways) entry.
+
+
+The gateways stanza is where you will define and configure the set of ingress gateways you want deployed to your environment.
+The only required field for each entry is `name`, though entries may contain any of the fields found in the `defaults` stanza.
+Values in this section override the values from the defaults stanza for the given ingress gateway with one exception:
+the annotations from the defaults stanza will be *appended* to any user-defined annotations defined in the gateways stanza rather than being overridden.
+Please refer to the ingress gateway configuration [documentation](/docs/k8s/helm#v-ingressgateways-defaults) for a detailed explanation of each option.
+
+-> *Note*: Make sure any ports that will be used as listeners in the ingress gateway's Consul config entry are included in the `ports` object for each gateway.
+
+## Deploying the helm chart
+
+Ensure you have the latest consul-helm chart and install Consul via helm using the following
+[guide](/docs/k8s/installation/overview#installing-consul) while being sure to provide the yaml configuration
+as previously discussed.
+
+## Configuring the gateway
+
+Now that Consul has been installed with ingress gateways enabled, you must add the corresponding configuration to Consul. This requires you to use the Consul CLI.
+Configuring the ingress gateway requires:
+
+* Accessing the Consul server
+* Submitting an Ingress Gateway configuration entry to Consul
+
+### Accessing the Consul server
+
+You can access the Consul server directly from your host via `kubectl port-forward`, this is helpful for interacting with the Consul UI locally as well as validating connectivity of the application.
+
+```shell-session
+$ kubectl port-forward consul-server-0 8500 &
+```
+-> If TLS is enabled use port 8501
+
+-> Download the latest Consul binary from [Downloads](/downloads.html).
+[https://releases.hashicorp.com/consul/](https://releases.hashicorp.com/consul/)
+
+-> If TLS is enabled set:
+```shell-session
+$ export CONSUL_HTTP_ADDR=https://localhost:8501
+```
+
+-> If ACLs are enabled set :
+```shell-session
+$ export CONSUL_HTTP_TOKEN=$(kubectl get secret consul-bootstrap-acl-token -o jsonpath={.data.token} | base64 -D)
+$ export CONSUL_HTTP_SSL_VERIFY=false
+```
+
+### Submitting an Ingress Gateway configuration entry
+
+Now that you have access to a Consul server through the forwarded port and you have the Consul CLI configured locally, you are able to submit an ingress gateway configuration entry.
```

Comment:
We should probably link this to the full [ingress gateway config entry](https://www.consul.io/docs/connect/ingress-gateway#ingress-gateway-configuration) docs page, though the meat of that is really in the [config entry](https://www.consul.io/docs/agent/config-entries) doc. The first is probably better since the second can be found from there.

Reply:
Ah, this is better: https://www.consul.io/docs/agent/config-entries/ingress-gateway